### PR TITLE
feat(storage-node): add reconcile logic

### PR DESCRIPTION
**Pull request by Xu-Wentao**
### Type of change:

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:

- implement storage node `reconcile` logic
- support `aws-aurora` database
### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?

### CODE REVIEW

**Comment by mlycore** [comment]
Please change the remote branch to `feat-storagenode`

**Review by mlycore — APPROVED** [review]

File: shardingsphere-operator/api/v1alpha1/storage_node_types.go [hunk 1]
```diff
@@ -17,7 +17,65 @@
...
+const (
+	StorageNodePhaseReady    StorageNodePhaseStatus = "Ready"
+	StorageNodePhaseNotReady StorageNodePhaseStatus = "NotReady"
+)
+
+type StorageNodeConditionType string
+
+// StorageNodeConditionType shows some states during the startup process of storage node
+const (
+	StorageNodeConditionTypeAvailable StorageNodeConditionType = "Available"
```
Comment (mlycore) [inline shardingsphere-operator/api/v1alpha1/storage_node_types.go:36]: Please add explanation for these two conditions and update related design proposal.

File: shardingsphere-operator/api/v1alpha1/storage_node_types.go [hunk 2]
```diff
@@ -17,7 +17,65 @@
...
+}
+
+type InstanceStatus struct {
+	Status     string            `json:"status"`
+	Endpoint   Endpoint          `json:"primaryEndpoint"`
+	Properties map[string]string `json:"properties"`
+}
+
+type BasicCredential struct {
+	Username string `json:"user"`
```
Comment (mlycore) [inline shardingsphere-operator/api/v1alpha1/storage_node_types.go:71]: For better maintainability, please tag the object member with same name.

**Review by mlycore — CHANGES_REQUESTED** [review]

**Comment by Xu-Wentao** [comment]
> Please change the remote branch to `feat-storagenode`

done!

File: shardingsphere-operator/api/v1alpha1/storage_node_types.go (hunk 1 above)
Comment (Xu-Wentao) [reply shardingsphere-operator/api/v1alpha1/storage_node_types.go:37]: Just retain the "Available" condition for describe the whole storageNode is available or not, it's depends cluster is ready and all instances are ready.
We may add more additional conditions for more info in the future.

Comment (mlycore) [reply shardingsphere-operator/api/v1alpha1/storage_node_types.go:37]: > Just retain the "Available" condition for describe the whole storageNode is available or not, it's depends cluster is ready and all instances are ready. We may add more additional conditions for more info in the future.

Yeah, this makes sense.

**Pull request merged**